python/tests/test_docs.py

```diff
@@ -0,0 +1,121 @@
+from konduit import *
+
+
+def markdown_blocks(file_path, language='python', safe=True):
+    code_blocks = []
+    code_regex = r'^```.*'
+    code_block_open_re = r'^```(`*)({0}){1}$'.format(language, '' if safe else '?')
+
+    with open(file_path, 'r') as f:
+        block = []
+        python = True
+        in_code_block = False
+
+        for line in f.readlines():
+            code_block_delimiter = re.match(code_regex, line)
+
+            if in_code_block:
+                if code_block_delimiter:
+                    if python:
+                        code_blocks.append(''.join(block))
+                    block = []
+                    python = True
+                    in_code_block = False
+                else:
+                    block.append(line)
+            elif code_block_delimiter:
+                in_code_block = True
+                if not re.match(code_block_open_re, line):
+                    python = False
+    return code_blocks
+
+
+def is_markdown(f):
+    markdown_extensions = ['.markdown', '.mdown', '.mkdn', '.mkd', '.md']
+    return os.path.splitext(f)[1] in markdown_extensions
+
+
+def get_files(input_dir):
+    return [os.path.join(dp, f) for dp, dn, file_names in os.walk(input_dir) for f in file_names if is_markdown(f)]
+
+
+def exec_python_code(code_blocks):
+    for block in code_blocks:
+        try:
+            exec(block)
+        except Exception as e:
+            print("Test execution failed in the following block:\n")
+            print(block)
+            print("failed with error message:\n")
+            raise e
+
+
+def make_folders():
+    if not os.path.isdir('./ai'):
+        os.mkdir('./ai')
+    if not os.path.isdir('./ai/konduit'):
+        os.mkdir('./ai/konduit')
+    if not os.path.isdir('./ai/konduit/serving'):
+        os.mkdir('./ai/konduit/serving')
+
+
+def write_java_files(code_blocks):
+    make_folders()
+    for i, block in enumerate(code_blocks):
+        lines = block.split('\n')
+        import_lines = [l for l in lines if l.startswith('import ')]
+        code_lines = [l for l in lines if not l.startswith('import ')]
+
+        code = "package ai.konduit.serving;\n\n" + '\n'.join(import_lines) \
```

Comment:
You could just wrap this in a main method and run it like that maybe? Also - what about maybe having a java file you load as a template we bundle instead? I know it might seem annoying but it would make this a bit more maintainable than stringly typed code :) That's what I did to the python executioner.

Reply:
@agibsonccc sure, let me think about a "main" runner (right now I add the class to the konduit.jar, so that you only need to do that once. unit tests run really fast like this).

well, about the template. you see that it's literally 5 lines of boiler plate, the rest comes from code snippets in docs. and those snippets _are_ strings, no matter how we build it in the end. also, keep in mind that I build one java class per code snippet to ensure they're not entangled and each snippet runs standalone as expected. not sure how I would to that with templates (I don't know how many I need, for starters)